p2p/libp2p/netMessenger.go

```diff
@@ -218,6 +219,17 @@ func createPubSub(ctxProvider *Libp2pContext, withSigning bool) (*pubsub.PubSub,
 	return ps, nil
 }
 
+// ApplyOptions can set up different configurable options of a networkMessenger instance
+func (netMes *networkMessenger) ApplyOptions(opts ...Option) error {
+	for _, opt := range opts {
+		err := opt(netMes)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
```

Comment:
I think the main idea behind using options is to simplify the constructor, and restrict applying them at "object" creation.

Otherwise we should just make the options regular setters .

Reply:
Ok, this is the first step for the future refactoring process. I can not simply change the netMessenger's constructor as it will make the PR longer.